homeassistant/components/gios/__init__.py

```diff
@@ -43,36 +44,27 @@
 
 async def async_unload_entry(hass, config_entry):
     """Unload a config entry."""
-    hass.data[DOMAIN][DATA_CLIENT].pop(config_entry.entry_id)
+    hass.data[DOMAIN].pop(config_entry.entry_id)
     await hass.config_entries.async_forward_entry_unload(config_entry, "air_quality")
     return True
 
 
-class GiosData:
+class GiosDataUpdateCoordinator(DataUpdateCoordinator):
     """Define an object to hold GIOS data."""
 
-    def __init__(self, session, station_id):
-        """Initialize."""
-        self._gios = Gios(station_id, session)
-        self.station_id = station_id
-        self.sensors = {}
-        self.latitude = None
-        self.longitude = None
-        self.station_name = None
-        self.available = True
-
-    @Throttle(DEFAULT_SCAN_INTERVAL)
-    async def async_update(self):
-        """Update GIOS data."""
+    def __init__(self, hass, session, station_id):
+        """Class to manage fetching GIOS data API."""
+        self.gios = Gios(station_id, session)
+
+        super().__init__(hass, _LOGGER, name=DOMAIN, update_interval=SCAN_INTERVAL)
+
+    async def _async_update_data(self):
+        """Update data via library."""
         try:
             with timeout(30):
```

Comment:
The timeout error isn't catched anymore now?

Reply:
I think DataUpdateCoordinator handles `asyncio.TimeoutError` exception:
https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/update_coordinator.py#L131

Reply:
Yep, we recently added that :)

Reply:
Oh right! 👍